### DISCUSSION TITLE
bafang bbshd get motor rpm?

### DISCUSSION THREAD

**Discussion by skilloo** [danielnilsson9/bbs-fw#201] [Ideas]
Hi,
Thanks @danielnilsson9  for your great work!

I'm developing mobile app with ability to display motor data / change firmware settings - I want to control motor entirely by phone and have detailed dashboard to monitor temp and all related motor stuff  (I'm using high voltage mod + shunt mod controller). 

I'm wonder if it possible to get from controller (with some adjustments/protocol additions in FW) current speed (RPM) of the motor? I found in src (bafang version) that I can get rpm of PAS sensor and SpeedSensor but nothing about motor speed - there is possibility to calculate this data from hall sensor somewhow? This can be great addition to motor dashboard

**Comment by danielnilsson9** [comment]
Hi

Sound interesting!

Yes, as far as I know hall sensors are also connected to STC MCU so should be pretty easy to calculate motor rpm. Check the drawings for which pins they are connected on.

I started adding a status message at some point for this type of data but never finished it, but I think some code is still there. Feel free to continue the work and send a PR to get it into the official version if you want.
